libraries/node-core-library/src/Import.ts

```diff
@@ -1,13 +1,58 @@
 // Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
 // See LICENSE in the project root for license information.
 
+import * as nodeJsPath from 'path';
 import importLazy = require('import-lazy');
+import * as Resolve from 'resolve';
+import nodeModule = require('module');
+
+import { PackageJsonLookup } from './PackageJsonLookup';
+import { FileSystem } from './FileSystem';
+
+/**
+ * @public
+ */
+export interface IImportResolveOptions {
+  /**
+   * The path to resolve.
+   */
+  resolvePath: string;
+
+  /**
+   * The path from which {@link IImportResolveOptions.resolvePath} should be resolved
+   */
+  baseFolderPath: string;
+
+  /**
+   * If true, if the package name matches a Node.js system module, then the return
+   * value will be the package name without any path.  This will take precedence over
```

Comment:
Use `@remarks` block to keep the summary from being too long.  The example will be more readable as a code block:

  /**
   * If true, if the package name matches a Node.js system module, then the return
   * value will be the package name without any path.  
   * 
   * @remarks
   * This will take precedence over an installed NPM package of the same name.
   *
   * Example:
   * ```ts
   * // Returns the string "fs" indicating the Node.js system module
   * Import.resolveModulePath({ 
   *   resolvePath: "fs", 
   *   basePath: process.cwd() 
   * })
   * ```
   */
  includeSystemModules?: boolean;

Reply:
Same for `allowSelfReference`